libs/providers/multi-provider/src/lib/multi-provider.ts

```diff
@@ -0,0 +1,402 @@
+import {
+  DefaultLogger,
+  EvaluationContext,
+  EventDetails,
+  FlagValueType,
+  GeneralError,
+  Hook,
+  HookContext,
+  HookHints,
+  JsonValue,
+  Logger,
+  OpenFeatureEventEmitter,
+  Provider,
+  ProviderEvents,
+  ProviderStatus,
+  ResolutionDetails,
+} from '@openfeature/server-sdk';
+import { BeforeHookContext, ProviderMetadata } from '@openfeature/core';
+import { BaseEvaluationStrategy, FirstMatchStrategy, ProviderResolutionResult } from './strategies';
+import { HookExecutor } from './hook-executor';
+import { AggregateError } from './errors';
+
+// Represents an entry in the constructor's provider array which may or may not have a name set
+export type ProviderEntryInput = {
+  provider: Provider;
+  name?: string;
+};
+
+// Represents a processed and "registered" provider entry where a name has been chosen
+type RegisteredProvider = {
+  name: string;
+  provider: Provider;
+};
+
+export class MultiProvider implements Provider {
+  readonly runsOn = 'server';
```

Comment:
I guess this brings up the question (probably fo the OF folks), should this be named the `ServerMultiProvider`?

Reply:
that wouldn't match the other provider naming conventions used in this repo, where an "unqualified" name seems to be assumed to mean "server", and web providers add "web" to the name